front-end/src/views/MessageForm.vue

```diff
@@ -182,20 +311,32 @@ const targetSettingsMessages = {
 };
 
 // xem preview
-const previewData = ref({
+interface PreviewData {
+  platform: string;
+  sender: string;
+  urlReceiver: string;
+  message: string;
+  targetSettings: string[];
+}
+
+const previewData: Ref<PreviewData> = ref({
   platform: "",
   sender: "",
-  receiverURL: "",
-  template: "",
+  urlReceiver: "",
   message: "",
-  targetSettings: [""],
+  targetSettings: [],
 });
 const showPreviewModel = ref(false);
 const preview = () => {
-  previewData.value.platform = selectedPlatform.value;
-  previewData.value.sender = selectedSender.value;
-  previewData.value.receiverURL = receiverURL.value;
-  previewData.value.template = selectedTemplate.value;
+  if (selectedPlatform.value)
+    previewData.value.platform = selectedPlatform.value.name;
```

Comment:
[Should] Không nên update property của một ref, thay vào đó sử dụng reactive.
Đôi lúc update property của một ref có thể không render lại UI (đặc biệt khi ref là một mảng)